module/zfs/vdev.c

```diff
@@ -1347,6 +1347,10 @@ vdev_open(vdev_t *vd)
 		vd->vdev_max_asize = max_asize;
 		if (vd->vdev_ashift == 0)
 			vd->vdev_ashift = ashift;
+		else
+			if (vd->vdev_ashift < ASHIFT_MIN ||
+			    vd->vdev_ashift > ASHIFT_MAX)
```

Comment:
Yes, i'm going to update the code as soon as i get back to my dev machine. Also, am i allowed to add `VDEV_AUX_BAD_ASHIFT` to the `vdev_aux_t` enum or should i use an existing value?